In [2]:
import sys
print(sys.executable)
print(sys.version)

c:\Users\sagri\AppData\Local\Programs\Python\Python314\python.exe
3.14.3 (tags/v3.14.3:323c59a, Feb  3 2026, 16:04:56) [MSC v.1944 64 bit (AMD64)]


In [1]:
from google.cloud import bigquery
import pandas as pd

# Use your actual GCP project ID throughout
PROJECT_ID = 'chrome-ability-449516-a4'

client = bigquery.Client(project=PROJECT_ID)

df_test = client.query("""
    SELECT * 
    FROM `chrome-ability-449516-a4.healthtrack_analytics.q1_funnel_summary`
    ORDER BY stage_order
""").to_dataframe()

print(df_test)

c:\Users\sagri\AppData\Local\Programs\Python\Python314\Lib\site-packages\google\auth\_default.py:114: UserWarning: Your application has authenticated using end user credentials from Google Cloud SDK without a quota project. You might receive a "quota exceeded" or "API not enabled" error. See the following page for troubleshooting: https://cloud.google.com/docs/authentication/adc-troubleshooting/user-creds. 
  warnings.warn(_CLOUD_SDK_CREDENTIALS_WARNING)
c:\Users\sagri\AppData\Local\Programs\Python\Python314\Lib\site-packages\google\cloud\bigquery\table.py:1994: UserWarning: BigQuery Storage module not found, fetch data with the REST endpoint instead.
  warnings.warn(


   stage_order           stage   users  conversion_rate_pct  dropoff_rate_pct  \
0            1   Session start  354857               100.00               NaN   
1            2     Viewed item   77020                21.70             78.30   
2            3   Added to cart   15188                 4.28             80.28   
3            4  Began checkout   11106                 3.13             26.88   
4            5       Purchased    4848                 1.37             56.35   

   sessions_lost  
0              0  
1         277837  
2          61832  
3           4082  
4           6258  


What this tells from business perspective:
1. Massive drop at top of funnel : 354k → 77k where only 21.7% even view a product.
- Poor landing page relevance
- Weak targeting or traffic quality

2. Biggest loss is at View → Add to cart : 77k → 15k, 80% drop-off
- Pricing not attractive
- Product page weak
- No urgency or trust signals

3. Checkout is relatively strong : 15k → 11k, only ~27% drop. This is actually good.

4. Final purchase drop is still high : 11k → 4.8k, 56% drop
- Payment friction
- Hidden costs
- Account creation barriers